### DISCUSSION TITLE
Smart apostrophes, also known as curly or typographer's apostrophes

### DISCUSSION THREAD

**Discussion by xixixao** [mdx-js/mdx#2268] [Q&A] [answered]
Is there a plugin for, or a code snippet to enable, converting single and double quotes to smart apostrophes, also known as curly or typographer's apostrophes?

`'` should convert to `’`
`You're` should convert to `You’re`

?

**Comment by ChristianMurphy** [comment, marked as answer]
Welcome @xixixao!
MDX is focused on markdown and JSX specifically.

If you're looking for natural language check out retext.
In particular https://github.com/retextjs/retext-quotes
MDX (through remark) can interface with retext using https://github.com/remarkjs/remark-retext

Or if you're looking for something more custom.
Check out https://unifiedjs.com/learn/
To learn more about working with syntax trees and creating your own plugin.

**Comment by wooorm** [comment]
> Is there a plugin […]?

My personal suggestion on this is that I believe authors should learn how to use their keyboard to input the correct characters.
Typography is important, but computers can’t do typography perfectly. Humans can!
